### ISSUE TITLE
Vec2 not serialized/iterated over properly.

### ISSUE THREAD

**Issue by Codingale** [Rusketh/ExpAdv2#357]
Doing:

    client {
        table Table;
        Scores.insert({owner().steamID(), vec()}) //vec
        printTable(Table)
    }

Works just fine but doing: 

    client {
        table Table;
        Scores.insert({owner().steamID(), vec2()}) //vec2
        printTable(Table)
    }

Causes this error:

`Expression Advanced 2 - Suffered a clientside Script error:
lua/includes/extensions/table.lua:711: bad argument #1 to 'pairs' (table expected, got nil)`

Vector is defaulted to a proper vector (`vector(0,0,0)`), where as vec2 is causing an error when it's attempted to print.

**Comment by bigdogmat** [comment]
Does anything print out before the error? If so the issue lies in how the printTable function works, shouldn't be too hard to fix, just needs proper type checking.

**Comment by Codingale** [comment]
It happens during the print table yes, at the part where it prints it. I see the SteamID. I've not coded much in Lua so I wasn't sure where the fault was.

**Comment by bigdogmat** [comment]
The fault is in the type checking. The function uses recursion to print nested tables, the issue here is that Vector2 is a custom implementation, and so it uses tables to store the data. The reason type checking is at fault is because the current check says that the Vector2 is a table, and so it tries to recursively print it, though because it's not a table (in terms of expadv2) it doesn't have the table structure, and so it errors while trying to get the Data key.

TLDR: The fault is in the type checking, currently it says that the Vector2 object is a table.

I'll make a pr it fix it once I'm home.
